File: 4/python/funcs.py
def is_palindromic(n):
    return n == int(str(n)[::-1])

def find_palindromic(n1, n2):
    '''
    finds highest product of two numbers that is palindromic in range of 1 to n1
    and 1 to n2
   '''
    result = 0
    for i in range(n1+1):
       for j in range(n2+1):
           prod = i*j
           if is_palindromic(prod) and prod > result:
               result = i * j
               first_n = i
               second_n = j
    return {"prod":result, "n1":first_n, "n2":second_n}

File: 4/JS/funcs.js
function reverse(n) {
    var reversed = "";
    for (i in String(n)) {
        reversed = String(n)[i].concat(reversed);
    }
    return Number(reversed);
};

function palindromic(n1, n2) {
    var prod;
    var result = 0;
    var biggest_n1;
    var biggest_n2;

    for (let i = 0; i < n1; i++) {
        for (let j = 0; j < n2; j++) {
            prod = i * j;
            if(prod > result && reverse(prod) === prod) {
                result = prod;
                biggest_n1 = i;
                biggest_n2 = j;
            }

        }
    }
    return {"prod": result, "n1":biggest_n1, "n2":biggest_n2}
}
